src/selectors_vm/parser.rs

```diff
@@ -49,6 +50,14 @@ impl NonTSPseudoClass for NonTSPseudoClassStub {
     fn is_active_or_hover(&self) -> bool {
         match *self {}
     }
+
+    fn is_user_action_state(&self) -> bool {
+        match *self {}
+    }
+
+    fn has_zero_specificity(&self) -> bool {
+        match *self {}
+    }
```

Comment:
I known this isn't new code  but how does it return a bool?

Reply:
This is an enum without any variants, it's impossible for it to be constructed. So if this function is executed, UB has already occurred, so the return type can be anything.

(If it helps, this type could be an alias for std:: Infallible, I'm not sure why it's not.)